File: API/ECommerceTest/Models/Order/OrderModel.cs
namespace ECommerceTest.Models.Order;

public record OrderPaymentModel(string type);

public record OrderDeliveryModel(string type, string destination);
public record OrderProductModel(string id, uint amount);

public record OrderModel(
	string id,
	OrderPaymentModel payment,
	OrderDeliveryModel delivery,
	OrderProductModel[] products
);

File: src/app/services/backend-models/order.model.ts
export enum OrderPaymentType { Cash, Card };
export enum OrderDeliveryType { Pickup, NovaPoshta, UkrPoshta };

export type OrderPaymentModel = { type: OrderPaymentType };
export type OrderDeliveryModel = {
	type: OrderDeliveryType,
	destination?: string
};

export type OrderProductModel = {
	id: string,
	amount: number
};
export type OrderModel = {
	payment: OrderPaymentModel,
	delivery: OrderDeliveryModel,
	products: OrderProductModel[]
};

export type OrderPlaceResultModel = {
	orderId: string,
	invoice: {
		id: string,
		url: string
	}
};
